python/vowpalwabbit/pyvw.py

```diff
@@ -1224,6 +1232,61 @@ def __init__(self, action: int, score: float):
         self.score = score
 
 
+class CCBSlotOutcome:
+    def __init__(self, cost: float = 0, action_probs: List[ActionScore] = [], **kwargs):
+        ex: example = kwargs.get("ex")
+        if isinstance(ex, example):
+            self.from_example(ex)
+        else:
+            self.cost = cost
+            self.action_probs = action_probs
+
+    def from_example(self, ex: "example"):
+        self.cost = ex.get_ccb_cost()
+        self.action_probs = []
+        for i in range(ex.get_ccb_num_included_actions()):
+            self.action_probs.append((ex.get_ccb_class(i), ex.get_ccb_probs(i)))
+
+    def __str__(self):
+        top_action, top_score = self.action_probs.action, self.action_probs.score
+        out = f"{top_action}:{top_score}:{self.cost}"
+        for action_score in self.action_probs[1:]:
+            out += f",{action_score.action}:{action_score.score}"
+        return out
+
+
+class CCBLabel(AbstractLabel):
+    """Class for conditional contextual bandits VW label"""
+
+    def __init__(
+        self,
+        type: CCBLabelType = CCBLabelType.UNSET,
+        explicit_included_actions: List[int] = [],
+        weight: float = 1,
+        outcome: Optional[CCBSlotOutcome] = None,
+        **kwargs,
+    ):
+        AbstractLabel.__init__(self)
+        ex: example = kwargs.get("ex")
+        if isinstance(ex, example):
+            if not ex.labelType == LabelType.CONDITIONAL_CONTEXTUAL_BANDIT:
+                raise ValueError("Example should have CCB label type.")
```

Comment:
I think other classes should follow this, or am I missing something?

Reply:
Yes this would be safer -- adding in a separate PR

Reply:
Okay, removing it (and the corresponding test) for the sake of maintaining scope.